rocAL/source/readers/image/coco_file_source_reader.cpp

```diff
@@ -197,30 +228,40 @@ int COCOFileSourceReader::release() {
 
 void COCOFileSourceReader::shuffle_with_aspect_ratios() {
     // Calculate the mid element which divides the aspect ratios into two groups (<=1.0 and >1.0)
-    auto mid = std::upper_bound(_aspect_ratios.begin(), _aspect_ratios.end(), 1.0f) - _aspect_ratios.begin();
+    auto shard_start_idx = _shard_start_idx_vector[_shard_id];
+    auto mid = std::upper_bound(_aspect_ratios.begin() + shard_start_idx, _aspect_ratios.begin() + shard_start_idx + actual_shard_size_without_padding(), 1.0f) - (_aspect_ratios.begin() + shard_start_idx);
```

Comment:
shard_start_idx + actual_shard_size_without_padding()
Can this be replaced with _shard_end_idx_vector, I am not sure please check

Reply:
_shard_end_idx_vector has the value shard_start_idx + actual_shard_size_without_padding() - 1
Need to check which is the right one